spark/src/main/java/org/apache/iceberg/spark/data/SparkOrcReader.java

```diff
@@ -0,0 +1,823 @@
+    @Override
+    public void convert(UnsafeRowWriter writer, int column, ColumnVector vector,
+                        int row) {
+      if (vector.isRepeating) {
+        row = 0;
+      }
+      if (!vector.noNulls && vector.isNull[row]) {
+        writer.setNullAt(column);
+      } else {
+        writer.write(column, ((DoubleColumnVector) vector).vector[row]);
+      }
+    }
+
+    @Override
+    public void convert(UnsafeArrayWriter writer, int element,
+                        ColumnVector vector, int row) {
+      if (vector.isRepeating) {
+        row = 0;
+      }
+      if (!vector.noNulls && vector.isNull[row]) {
+        writer.setNull(element);
+      } else {
+        writer.write(element, ((DoubleColumnVector) vector).vector[row]);
+      }
+    }
+  }
+
+  private static class TimestampConverter implements Converter {
+
+    private long convert(TimestampColumnVector vector, int row) {
+      // compute microseconds past 1970.
+      long micros = (vector.time[row]/1000) * 1_000_000 + vector.nanos[row] / 1000;
+      return micros;
+    }
+
+    @Override
+    public void convert(UnsafeRowWriter writer, int column, ColumnVector vector,
+                        int row) {
+      if (vector.isRepeating) {
+        row = 0;
+      }
+      if (!vector.noNulls && vector.isNull[row]) {
+        writer.setNullAt(column);
+      } else {
+        writer.write(column, convert((TimestampColumnVector) vector, row));
+      }
+    }
+
+    @Override
+    public void convert(UnsafeArrayWriter writer, int element,
+                        ColumnVector vector, int row) {
+      if (vector.isRepeating) {
+        row = 0;
+      }
+      if (!vector.noNulls && vector.isNull[row]) {
+        writer.setNull(element);
+      } else {
+        writer.write(element, convert((TimestampColumnVector) vector, row));
+      }
+    }
+  }
+
+  private static class BinaryConverter implements Converter {
+
+    @Override
+    public void convert(UnsafeRowWriter writer, int column, ColumnVector vector, int row) {
+      if (vector.isRepeating) {
+        row = 0;
+      }
+      if (!vector.noNulls && vector.isNull[row]) {
+        writer.setNullAt(column);
+      } else {
+        BytesColumnVector v = (BytesColumnVector) vector;
+        writer.write(column, v.vector[row], v.start[row], v.length[row]);
+      }
+    }
+
+    @Override
+    public void convert(UnsafeArrayWriter writer, int element, ColumnVector vector, int row) {
+      if (vector.isRepeating) {
+        row = 0;
+      }
+      if (!vector.noNulls && vector.isNull[row]) {
+        writer.setNull(element);
+      } else {
+        final BytesColumnVector v = (BytesColumnVector) vector;
+        writer.write(element, v.vector[row], v.start[row], v.length[row]);
+      }
+    }
+  }
+
+  /**
+   * This hack is to get the unscaled value (for precision <= 18) quickly.
+   * This can be replaced when we upgrade to storage-api 2.5.0.
```

Comment:
Is storage-api 2.5.0 available?

Reply:
@edgarRd, did the update to ORC 1.5.0 fix this?

Reply:
I used `serialize64` which wraps `fastSerialize64` for fast serialization.